src/player_instances.c

```diff
@@ -237,10 +237,10 @@ long pinstfe_hand_whip(struct PlayerInfo *player, long *n)
   }
   case TCls_Shot:
       shotst = get_shot_model_stats(thing->model);
-      if (shotst->model_flags & ShMF_Boulder)
+      if ((shotst->model_flags & ShMF_Boulder) || (shotst->model_flags & ShMF_Slappable))
```

Comment:
if it got to this point it means it is slappable already, given it passed the thing_slappable, at the top of pinstfe_hand_whip, here it needs to determine whether a slap should detonate the shot, or redirect it, so checking on slappable here makes no sense, you made everything redirect